python_modules/dagster/dagster/_core/definitions/asset_layer.py

```diff
@@ -633,6 +633,10 @@ def asset_keys(self) -> Iterable[AssetKey]:
     def has_assets_defs(self) -> bool:
         return len(self.assets_defs_by_key) > 0
 
+    @property
+    def assets_defs(self) -> Set["AssetsDefinition"]:
+        return set(assets_def for assets_def in self.assets_defs_by_key.values())
+
```

Comment:
seems expensive enough to warrant `cached_property`

Reply:
I don't think its too expensive to calculate since its just doing `id` for hash, but also the set just holds a bunch of pointers so the memory cost is probably negligible. 

Since its unclear, my intuition was to wait til it measures as problematic.

Reply:
fair enough. just wanted to flag in case it was a mistake/oversight.